exporters/autoexport/metrics.go

```diff
@@ -94,4 +99,47 @@ func init() {
 	RegisterMetricReader("none", func(ctx context.Context) (metric.Reader, error) {
 		return newNoopMetricReader(), nil
 	})
+	RegisterMetricReader("prometheus", func(ctx context.Context) (metric.Reader, error) {
+		host := os.Getenv("OTEL_EXPORTER_PROMETHEUS_HOST")
+		if host == "" {
+			host = "localhost"
+		}
+
+		port := os.Getenv("OTEL_EXPORTER_PROMETHEUS_PORT")
+		if port == "" {
+			port = "9464"
+		}
+
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.Handler())
```

Comment:
What do you think of creating an explicit registerer that we can pass using `WithRegisterer` option)? I think it could be safer than using the globals offered by prometheus package if for any reason the user would already use the prometheus globals.

See: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#hdr-A_Basic_Example

@dashpole WDYT?

Reply:
That makes sense to me.  It means you won't get the "default" prometheus instrumentation experience, like go runtime metrics, but that is probably OK, since OTel will have that kind of instrumentation available.

It does make it harder to combine prometheus instrumentation with otel instrumentation in a single endpoint.  But if someone needs to do that, autoexport might not be the best solution for them...

Reply:
Done.